apps/pipelines/tests/test_runnable_builder.py

```diff
@@ -1220,3 +1222,38 @@ def _tool_call(route):
     assert output["messages"][-1] == "D d"
     output = runnable.invoke(PipelineState(messages=["z"], experiment_session=experiment_session))
     assert output["messages"][-1] == "A z"
+
+
+@pytest.mark.django_db()
+def test_router_node_output_structure(provider, provider_model, pipeline, experiment_session):
+    service = build_fake_llm_echo_service()
+    with mock.patch("apps.service_providers.models.LlmProvider.get_llm_service", return_value=service):
+        node = RouterNode(
+            name="test_router",
+            prompt="PD: {participant_data}",
+            keywords=["A"],
+            llm_provider_id=provider.id,
+            llm_provider_model_id=provider_model.id,
+        )
+        state = PipelineState(
+            outputs={"123": {"message": "hello world"}},
+            messages=["hello world"],
+            experiment_session=experiment_session,
+            temp_state={"user_input": "hello world", "outputs": {}},
+            path=[],
+        )
+        with mock.patch.object(node, "_process_conditional", return_value="A"):
+            node_id = "123"
+            edge_map = {"A": "next_node_a", "B": "next_node_b"}
+            incoming_edges = ["123"]
+            router_func = node.build_router_function(node_id, edge_map, incoming_edges)
+            command = router_func(state, {"metadata": {"langgraph_triggers": []}})
```

Comment:
nit: the metadata is unnecessary
```suggestion
            command = router_func(state, {})
```